services/dataAccessLayer.js

```diff
@@ -0,0 +1,78 @@
+const userQuery = require("../models/users");
+const { getPaginationLink } = require("../utils/users");
+const { SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
+const { getQualifiers } = require("../utils/helper");
+const { getUsernamesFromPRs } = require("../utils/users");
+const { getFilteredPRsOrIssues } = require("../utils/pullRequests");
+
+const retrieveUsers = async (req, res) => {
+  try {
+    // getting user details by id if present.
+    const query = req.query?.query ?? "";
+    const qualifiers = getQualifiers(query);
+
+    // getting user details by id if present.
+    if (req.query.id) {
+      const id = req.query.id;
+      let result;
+      try {
+        result = await userQuery.fetchUser({ userId: id });
+      } catch (error) {
+        logger.error(`Error while fetching user: ${error}`);
+        return res.boom.serverUnavailable(SOMETHING_WENT_WRONG);
+      }
+
+      if (!result.userExists) {
+        return res.boom.notFound("User doesn't exist");
+      }
+
+      const user = result.user;
+      removeSensitiveInfo(user, ["phone", "email", "chaincode", "tokens"]);
+
+      return res.json({
+        message: "User returned successfully!",
+        user,
+      });
+    }
+
+    if (qualifiers?.filterBy) {
+      const allPRs = await getFilteredPRsOrIssues(qualifiers);
+      const usernames = getUsernamesFromPRs(allPRs);
+      const { users } = await userQuery.fetchUsers(usernames);
+      removeSensitiveInfo(users, ["phone", "email", "chaincode", "tokens"]);
+
+      return res.json({
+        message: "Users returned successfully!",
+        users,
+      });
+    }
+
+    const { allUsers, nextId, prevId } = await userQuery.fetchPaginatedUsers(req.query);
+    allUsers.forEach((element) => {
+      removeSensitiveInfo(element, ["phone", "email", "chaincode", "tokens"]);
+    });
+    return res.json({
+      message: "Users returned successfully!",
+      users: allUsers,
+      links: {
+        next: nextId ? getPaginationLink(req.query, "next", nextId) : "",
+        prev: prevId ? getPaginationLink(req.query, "prev", prevId) : "",
+      },
+    });
+  } catch (error) {
+    logger.error(`Error while fetching all users: ${error}`);
+    return res.boom.serverUnavailable(SOMETHING_WENT_WRONG);
+  }
+};
+
+const removeSensitiveInfo = function (obj, props) {
```

Comment:
A suggestion here

Can we have the function definition as given below?
This way, you don't have to pass an array every time you want to call the removeSensitiveInfo function, and it also gives you the flexibility to change parameters when required by passing an array as the second parameter.

```
function removeSensitiveInfo(data, properties = ["phone", "email", "chaincode", "tokens"]) {
  // Logic to remove sensitive information based on the properties array
  // ...
}
```

Example: 
```
 removeSensitiveInfo(element) // removes "phone", "email", "chaincode", and  "tokens"  as the default behavior
 removeSensitiveInfo(element, [token, chaincode])  // For cases where tokens and chaincode need to be removed.
```